beetsplug/vocadb.py

```diff
@@ -15,52 +21,175 @@
     name: str
     base_url: str
     api_url: str
     subcommand: str
 
 
+class APIResultDict(TypedDict):
+    id: NotRequired[int]
+
+
+class ArtistDict(APIResultDict):
+    additionalNames: str
+    artistType: str
+    deleted: bool
+    name: NotRequired[str]
+    pictureMime: str
+    status: str
+    version: int
+
+
+class AlbumArtistDict(APIResultDict):
+    artist: NotRequired[ArtistDict]
+    categories: str
+    effectiveRoles: str
+    isSupport: bool
+    name: NotRequired[str]
+    roles: str
+
+
+class TagDict(APIResultDict):
+    additionalNames: NotRequired[str]
+    categoryName: NotRequired[str]
+    name: str
+    urlSlug: NotRequired[str]
+
+
+class TagUsageDict(TypedDict):
+    count: int
+    tag: TagDict
+
+
+class InfoDict(APIResultDict):
+    artists: NotRequired[list[AlbumArtistDict]]
+    artistString: NotRequired[str]
+    createDate: NotRequired[str]
+    defaultName: NotRequired[str]
+    defaultNameLanguage: NotRequired[str]
+    name: NotRequired[str]
+    status: NotRequired[str]
+    tags: NotRequired[list[TagUsageDict]]
+
+
+class LyricsDict(APIResultDict):
+    cultureCodes: list[str]
+    source: NotRequired[str]
+    translationType: str
+    url: NotRequired[str]
+    value: str
+
+
+class DiscDict(APIResultDict):
+    discNumber: int
+    mediaType: str
+    name: NotRequired[str]
+    total: NotRequired[int]
+
+
+class ReleaseDateDict(TypedDict):
+    day: NotRequired[int]
+    isEmpty: NotRequired[bool]
+    month: NotRequired[int]
+    year: NotRequired[int]
+
+
+class SongDict(InfoDict):
+    favoritedTimes: int
+    lengthSeconds: int
+    lyrics: list[LyricsDict]
+    maxMilliBpm: int
+    minMilliBpm: int
+    publishDate: str
+    pvServices: str
+    ratingScore: int
+    songType: str
+    version: int
+    cultureCodes: list[str]
+
+
+class SongInAlbumDict(APIResultDict):
+    discNumber: int
+    name: NotRequired[str]
+    song: SongDict
+    trackNumber: int
+    computedCultureCodes: list[str]
+
+
+class WebLinkDict(APIResultDict):
+    category: str
+    description: str
+    descriptionOrUrl: str
+    disabled: bool
+    url: str
+
+
+class AlbumDict(InfoDict):
+    catalogNumber: NotRequired[str]
+    discs: Sequence[NotRequired[DiscDict]]
+    discType: NotRequired[str]
+    releaseDate: NotRequired[ReleaseDateDict]
+    tracks: list[SongInAlbumDict]
+    webLinks: NotRequired[list[WebLinkDict]]
+
+
+class FindResultDict(APIResultDict):
+    term: str
+    totalCount: int
+
+
+class SongFindResultDict(FindResultDict):
+    items: list[SongDict]
+
+
+class AlbumFindResultDict(FindResultDict):
+    items: list[AlbumDict]
+
+
 class VocaDBPlugin(BeetsPlugin):
 
     user_agent: str = f"beets/{beets.__version__} +https://beets.io/"
     headers: dict[str, str] = {"accept": "application/json", "User-Agent": user_agent}
-    languages = config["import"]["languages"].as_str_seq()
+    languages: list[str] | None = [config["import"]["languages"].as_str_seq()]
```

Comment:
```suggestion
    languages: Union[list[str], None] = [config["import"]["languages"].as_str_seq()]
```
This could also be Optional (though maybe you removed uses of Optional for a reason?):
```suggestion
    languages: Optional[list[str]] = [config["import"]["languages"].as_str_seq()]
```